385/Assets/Scripts/PlayerCamera.cs

```diff
@@ -31,12 +43,24 @@ public class PlayerCamera : MonoBehaviour {
     // updates camera button and camera position
 	void LateUpdate () {
         CameraButton.Update();
-        transform.position = player.transform.position + offset;
 
         // checks if camera button has been pressed
         if (CameraButton.IsButtonClicked())
         {
-            playerCam.enabled = !playerCam.enabled;
+            trackPlayer = !trackPlayer;
+        }
+
+        if (trackPlayer)
+        {
+            float step = moveSpeed * Time.deltaTime;
+            Vector3 temp = player.transform.position + offset;
+            transform.position = Vector3.MoveTowards(transform.position,temp, step);
```

Comment:
TIL MoveTowards is a thing. Looks like a good solution to address smooth camera movement. Good find 👍